v2/cli/project_roles_commands.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/brigadecore/brigade/sdk/v2/authz"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	libAuthz "github.com/brigadecore/brigade/sdk/v2/lib/authz"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	projectRoleGrantFlags = []cli.Flag{
+		&cli.StringFlag{
+			Name:     flagID,
+			Aliases:  []string{"i", flagProject, "p"},
+			Usage:    "Grant the role for the specified project (required)",
+			Required: true,
+		},
+		&cli.StringSliceFlag{
```

Comment:
For this and similar commands, had we considered a single `--principal` flag to provide the principal ID (which could be a user or service account) instead of the two `-u` and `-s` flags?

Reply:
If we did that, we'd need some way (another flag) to disambiguate whether the principal in question is a user or a service account. If we keep the two separate flags, that context is intrinsic.